poetry/repositories/installed_repository.py

```diff
@@ -49,20 +51,17 @@ def load(cls, env):  # type: (Env) -> InstalledRepository
                     is_standard_package = False
 
                 if is_standard_package:
-                    if (
-                        path.name.endswith(".dist-info")
-                        and env.site_packages.joinpath(
-                            "{}.pth".format(package.pretty_name)
-                        ).exists()
-                    ):
-                        with env.site_packages.joinpath(
-                            "{}.pth".format(package.pretty_name)
-                        ).open() as f:
-                            directory = Path(f.readline().strip())
-                            package.source_type = "directory"
-                            package.source_url = directory.as_posix()
-
-                    continue
+                    directories = set()
+                    site.addpackage(
+                        env.site_packages.as_posix(),
+                        "{}.pth".format(package.pretty_name),
+                        directories,
+                    )
+                    if directories:
+                        # TODO: handle multiple source directories
+                        package.source_type = "directory"
+                        package.source_url = directories.pop()
+                        continue
```

Comment:
```suggestion
                    try:
                        pth_file = env.site_packages.joinpath("{}.pth".format(package.pretty_name)).resolve(strict=True)
                        with pth_file.open() as f:
                            for line in f:
                                line = line.strip()
                                if not line or line.startswith(("#", "import ", "import\t")):
                                    # we ignore executables and comments
                                    continue
    
                                # we expect this line to be either a relative or an absolute path
                                path = Path(line)
    
                                if not path.is_absolute():
                                    path = env.site_packages.joinpath(path)
    
                                if path.is_dir():
                                    # TODO: handle multiple source directories?
                                    package.source_type = "directory"
                                    package.source_url = path.as_posix()
                                    break
                    except FileNotFoundError:
                        pass
```
It looks like the `addpackage` method has as a side-effect that is undesirable here.